tstest/tlstest/tlstest.go

```diff
@@ -0,0 +1,70 @@
+// Copyright (c) Tailscale Inc & AUTHORS
+// SPDX-License-Identifier: BSD-3-Clause
+
+// Package tlstest contains code to help test Tailscale's client proxy support.
+package tlstest
+
+import (
+	"crypto/tls"
+	_ "embed"
+)
+
+// TestRootCA is a self-signed ECDSA root CA certificate for testing
+// purposes.
+//
+//go:embed root-ca.crt
+var TestRootCA []byte
+
+// TestProxyServerKey is the private key for [TestProxyServerCert].
+//
+//go:embed proxy.tstest.key
+var TestProxyServerKey []byte
+
+// TestProxyServerCert is a server cert for the "proxy.tstest" domain
+// name signed by [TestRootCA].
+//
+//go:embed proxy.tstest.crt
+var TestProxyServerCert []byte
+
+// TestControlPlaneKey is the private key for [TestControlPlaneCert].
+//
+// It was generated from:
+//
+//go:embed controlplane.tstest.key
+var TestControlPlaneKey []byte
+
+// TestControlPlaneCert is a server cert for the "controlplane.tstest" domain
+// name signed by [TestRootCA].
+//
+//go:embed controlplane.tstest.crt
+var TestControlPlaneCert []byte
```

Comment:
these checked-in certs will eventually expire.
WDYT about generating them via `func init` and never storing on disk?